File: backend/Contracts/Responses/HealthCheckRes.cs
using Server.Contracts.Dtos;

namespace Server.Contracts.Responses;

public class HealthCheckRes
{
    public string Status { get; set; } = default!;

    public IEnumerable<HealthCheckDto> Checks { get; set; } = Enumerable.Empty<HealthCheckDto>();

    public TimeSpan Duration { get; set; }
}

File: frontend/lib/api/res/HealthCheckRes.ts
import { HealthCheckDtoSchema } from "@api/dtos/HealthCheckDto";
import { z } from "zod";

const HealthCheckResSchema = z.object({
  status: z.string(),
  checks: z.array(HealthCheckDtoSchema),
  duration: z.string(),
});

type HealthCheckRes = z.infer<typeof HealthCheckResSchema>;

export { HealthCheckResSchema };
export type { HealthCheckRes };
